### ISSUE TITLE
Persist messages in Volume

### ISSUE THREAD

**Issue by drummer3333** [lovelysystems/rabbitmq-swarm-cluster#5]
I like the concept of this, but I need to persist the messages to disk with a volume.

so I added a volume:
```yaml
    volumes:
       - /c/volumes/rabbit:/var/lib/rabbitmq
```

and changed the mnesia dir to be named after each slot:
```
      - "RABBITMQ_MNESIA_DIR=/var/lib/rabbitmq/mnesia/{{.Service.Name}}.{{.Task.Slot}}"
```

killing a single task is no problem.
But if I restart all tasks, all tasks want to reconnect to the old peers which are not avalible anymore.

**Comment by devavanza** [comment]
Hi @drummer3333,

Were you able to resolve it? Can you please share the workaround for adding persistence in this image?

**Comment by drummer3333** [comment]
no, I'm not using rabbit cluster mode in the moment. didn't have a problem till now.

**Comment by devavanza** [comment]
So you are using a single instance rabbitmq?

**Comment by drummer3333** [comment]
running in docker swarm, yes.

**Comment by TJFG1998** [comment]
Hey, using "RABBITMQ_MNESIA_DIR=/var/lib/rabbitmq/mnesia/{{.Service.Name}}.{{.Task.Slot}}.{{.Task.ID}}" should solve your problem.

**Comment by TJFG1998** [comment]
Try `hostname: "{{.Service.Name}}.{{.Task.Slot}}"`.
